signatureverifier/src/main/kotlin/io/github/rakutentech/signatureverifier/verification/SharedPreferenceEncryptor.kt

```diff
@@ -0,0 +1,50 @@
+package io.github.rakutentech.signatureverifier.verification
+
+import android.content.Context
+import android.os.Build
+import android.security.keystore.KeyGenParameterSpec
+import android.security.keystore.KeyProperties
+import androidx.security.crypto.EncryptedSharedPreferences
+import androidx.security.crypto.MasterKey
+
+internal class SharedPreferenceEncryptor(
+    context: Context,
+    keyGenerator: PreferenceEncryptorKeyGenerator = PreferenceEncryptorKeyGenerator(KEYSTORE_ALIAS),
+    pref: EncryptedSharedPreferences? = null
+) {
+
+    private val sharedPref = pref ?: EncryptedSharedPreferences.create(
+        context,
+        "signature_verifier_prefs",
+        keyGenerator.generateKey(context),
+        EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
+        EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
+    )
+
+    fun putEncryptedKey(keyId: String, key: String?) {
+        sharedPref.edit().putString(keyId, key).apply()
+    }
+
+    fun getEncryptedKey(keyId: String) = sharedPref.getString(keyId, null)
+
+    companion object {
+        private const val KEYSTORE_ALIAS = "signature-verifier-public-key-encryption-decryption"
+    }
+}
+
+internal class PreferenceEncryptorKeyGenerator(private val alias: String) {
+    fun generateKey(context: Context) =
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            val spec = KeyGenParameterSpec.Builder(
```

Comment:
Is there any advantage to manually defining a `KeyGenParameterSpec` vs using a default like:
`MasterKey.Builder(context).setKeyScheme(MasterKey.KeyScheme.AES256_GCM).build()`

Reply:
The only difference is the specs/settings is more customizable when using `KeyGenParameterSpec`. But you're right, using `setKeyScheme` is enough.
Updated this.